modules/gapi/doc/00-root.markdown

```diff
@@ -0,0 +1,143 @@
+# Graph API {#gapi}
+
+# Introduction {#gapi_root_intro}
+
+OpenCV Graph API (or G-API) is a new OpenCV module targeted to make
+regular image processing fast and portable. These two goals are
+achieved by introducing a new graph-based model of execution.
+
+G-API is a special module in OpenCV -- in contrast with the majority
+of other main modules, this one acts as a framework rather than some
+specific CV algorithm. G-API provides means to define CV operations,
+construct graphs (in form of expressions) using it, and finally
+implement and run the operations for a particular backend.
+
+# Contents
+
+G-API documentation is organized into the following chapters:
+
+- @subpage gapi_purposes
+
+  The motivation behind G-API and its goals.
+
+- @subpage gapi_hld
+
+  General overview of G-API architecture and its major internal
+  components.
+
+- @subpage gapi_kernel_api
+
+  Learn how to introduce new operations in G-API and implement it for
+  various backends.
+
+- @subpage gapi_impl
+
+  Low-level implementation details of G-API, for those who want to
+  contribute.
+
+- API Reference: functions and classes
+
+    - @subpage gapi_core
+
+      Core G-API operations - arithmetic, boolean, and other matrix
+      operations;
+
+    - @subpage gapi_imgproc
+
+      Image processing functions: color space conversions, various
+      filters, etc.
+
+# API Example {#gapi_example}
+
+A very basic example of G-API pipeline is shown below:
+
+@code{.cpp}
+    #include <opencv2/opencv.hpp>
+    #include <opencv2/gapi.hpp>
```

Comment:
It is better to extract tutorials code into separate files:
- create gapi folder here: https://github.com/opencv/opencv/tree/4.0.0-beta/samples/cpp/tutorial_code (or here: `modules/gapi/docs/code` - for better compatibility with standalone gapi)
- this code is checked at least for compilation
- use `@example` or `@snippet` to reference from documentation part.

Reply:
This is not a tutorial, this is an [almost useless] API sample -- just for an introduction.

I will add a more detailed tutorial into a proper location as it gets ready.

Reply:
Ah got it, will export the code into a distinct location, thanks!